packages/angular_devkit/build_angular/src/tools/esbuild/external-packages-plugin.ts

```diff
@@ -34,49 +32,31 @@ export function createExternalPackagesPlugin(options?: { exclude?: string[] }):
         return;
       }
 
-      const loaderFileExtensions = new Set(loaderOptionKeys);
+      const loaderFileExtensions = loaderOptionKeys && new Set(loaderOptionKeys);
 
-      // Only attempt resolve of non-relative and non-absolute paths
+      // esbuild's external packages option considers any package-like path as external
+      // which would exclude paths that are absolute or explicitly relative
       build.onResolve({ filter: /^[^./]/ }, async (args) => {
```

Comment:
Could this https://github.com/evanw/esbuild/pull/3667 help to exclude this callback to run already in esbuild? Like 

```ts
const loaders = loaderFileExtensions ? `(Array.from(loaderFileExtensions).join('|'))$` : '';
const libraries = '^(@angular|rxjs)\/)`';
const exclude = loaders ? `(${libraries}|${loaders})` : libraries;

build.onResolve({ filter: /^[^./]/, exclude }, async (args) => {
```

Reply:
No particularly for this use-case.

Reply:
Seems like my esbuild PR is not happening. At least I got no feedback for 3 months. No urgency on my part, as we patch angular cli via `yarn patch` to apply https://github.com/angular/angular-cli/pull/27120 ourself.